walletcontroller/client.go

```diff
@@ -75,6 +77,7 @@ func NewRpcWalletControllerFromArgs(
 		// we use post mode as it sure it works with either bitcoind or btcwallet
 		// we may need to re-consider it later if we need any notifications
 		HTTPPostMode: true,
+		Params:       params.Name,
```

Comment:
Not sure if this was needed, but I saw during debug that it was going to the fallthrough option if not set
```go
 // btcd/rpcclient/infrastructure.go

	// Default network is mainnet, no parameters are necessary but if mainnet
	// is specified it will be the param
	switch config.Params {
	case "":
		fallthrough
	case chaincfg.MainNetParams.Name:
		client.chainParams = &chaincfg.MainNetParams
	case chaincfg.TestNet3Params.Name:
		client.chainParams = &chaincfg.TestNet3Params
	case chaincfg.RegressionNetParams.Name:
		client.chainParams = &chaincfg.RegressionNetParams
	case chaincfg.SimNetParams.Name:
		client.chainParams = &chaincfg.SimNetParams
	default:
		return nil, fmt.Errorf("rpcclient.New: Unknown chain %s", config.Params)
	}
```

Reply:
yea, the problem is the upstream `btcd`, does not support `signet`. So if:
- we do not specify it then for `signet` we will run with `&chaincfg.MainNetParams` which for most cases is fine. Though it may, have some weird behaviour when decoding/encoding addresses
- if we do specify it then for `signet` staker will not start

For now in most of our programs, we do not specify it so that we are always able to start the programs on signet.

Reply:
removed at https://github.com/babylonlabs-io/btc-staker/pull/25/commits/c6d6b37731eb0e3f54cb451399f6c6d2202f27ef